packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -0,0 +1,588 @@
+                    previousTracesData[index].y,
+                    chartController.yPositonToYCanvasCoord(0),
+                    currentTracesData[index].x,
+                    currentTracesData[index].y,
+                    chartController.yPositonToYCanvasCoord(0),
+                  );
+                } else {
+                  // Draw point
+                  drawCircle(canvas, trace.characteristics, xCoord, yCoord);
+                }
+              }
+              tracesDataIndex[index]--;
+            }
+          }
+        }
+
+        previousTracesData.addAll(currentTracesData);
+        currentTracesData.clear();
+      }
+
+      // Undo the clipRect at beginning of for loop.
+      canvas.restore();
+
+      /////////////////////////////////////////////////////////////////////////
+      // Compute the ticks to label.
+      /////////////////////////////////////////////////////////////////////////
+
+      final minorTick = tickIndex++ < tickMarkInterval;
+
+      // Compute major tick labels
+      if (!minorTick) {
+        chartController.recomputeLabels(
+          timestamp: chartController.timestamps.last,
+        );
+      }
+
+      // Draw right-most tick label (first major tick).
+      drawXTick(
+        canvas,
+        currentTimestamp,
+        chartController.timestampXCanvasCoord(currentTimestamp),
+        axis,
+        shortTick: minorTick,
+      );
+
+      if (!minorTick) tickIndex = 0;
+    }
+
+    // Draw the major labels.
+    for (var labelIndex = 0;
+        labelIndex < chartController.getLabelsCount();
+        labelIndex++) {
+      final timestamp = chartController.getLabelTimestampByIndex(labelIndex);
+      if (timestamp != null) {
+        final xCoord = chartController.timestampXCanvasCoord(timestamp);
+        drawXTick(canvas, timestamp, xCoord, axis, displayTime: true);
+      }
+    }
+
+    // Rescale Y-axis to max visible Y range.
+    chartController.resetYMaxValue(visibleYMax);
+
+    // Draw Y-axis ticks and labels.
+    if (chartController.displayYLabels) {
+      drawYTicks(canvas, chartController, axis);
+    }
+
+    drawTitle(canvas, size, chartController.title);
+  }
+
+  void clipChart(Canvas canvas, {ClipOp op = ClipOp.intersect}) {
+    final leftSideSide = chartController.xCanvasChart;
+    final topChartSide = chartController.yCanvasChart;
+    final r = Rect.fromLTRB(
+      leftSideSide,
+      topChartSide,
+      chartController.canvasChartWidth +
+          leftSideSide -
+          chartController.xPaddingRight,
+      topChartSide + chartController.canvasChartHeight,
+    );
+
+    canvas.clipRect(r, clipOp: op);
+  }
+
+  // TODO(terry): Use drawText?
+  void drawTitle(Canvas canvas, Size size, String title) {
+    final tp = createText(title, 1.5);
+    tp.paint(canvas, Offset(size.width / 2 - tp.width / 2, 0));
+  }
+
+  void drawAxes(Canvas canvas, Size size, Paint axis) {
+    // Top line of chart.
+    canvas.drawLine(
```

Comment:
I'm seeing a lot of +xCanvasChart and +yCanvasChart here
you can 
write
https://api.flutter.dev/flutter/dart-ui/Canvas/translate.html
`canvas.translate(xCanvasChart, yCanvasChart)`
to simplify this math.

Reply:
Done.